src/modm/platform/core/cortex/vectors.c.in

```diff
@@ -33,7 +33,7 @@ typedef void (* const FunctionPointer)(void);
 extern uint32_t __main_stack_top[];
 
 // Define the vector table
-modm_section(".vector_rom")
+modm_section(".vector_rom") modm_used
```

Comment:
probably also necessary for `vectorsRam` just below.

Basically everything that's implicitly linked needs to be marked as used? Seems like a lot of work…

Reply:
Can you perhaps extract the symbols of the resulting ELF files with/without LTO and diff them? That would perhaps give you a more definitive list of lost symbols.
It's `scons symbols` but we seems to not have it for CMake.
Otherwise try: `arm-none-eabi-nm -S -C --size-sort -td file.elf`

Reply:
I don't see much difference any more. vectorsRam indeed is required to be used, I just use the ROM-mapped vector table in my project.

One more symbol missing is modm_abandon, but I think it is just optimised out as an empty function.

Reply:
Maybe, Undefined_Handler also requires the attribute. But I am not 100% sure

Reply:
I'll have a look myself tomorrow to check if we can enable LTO by default at least for the release build.